File: server/engines/c_cpp/defaults/defaults.cpp
// Test command line arguments and stdin
#include <stdio.h>

int main( int argc, char *argv[] )  {
    printf("Command line input: %s\n", argv[1]);
    
    char input[20];
    scanf("%[^\n]%*c", input);
    
    printf("Scanned input: %s\n", input);
} 

File: server/engines/java/defaults/Defaults.java
// Test command line arguments and stdin
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        System.out.println("Command line input: " + args[0]);
        
        Scanner Keyboard = new Scanner(System.in);
        String input = Keyboard.next();
        
        System.out.println("Scanned input: " + input);
    }
}